dbt_semantic_interfaces/validations/metrics.py

```diff
@@ -153,3 +154,113 @@ def validate_manifest(semantic_manifest: SemanticManifestT) -> Sequence[Validati
             issues += DerivedMetricRule._validate_alias_collision(metric=metric)
             issues += DerivedMetricRule._validate_time_offset_params(metric=metric)
         return issues
+
+
+class WhereFiltersAreParseable(SemanticManifestValidationRule[SemanticManifestT], Generic[SemanticManifestT]):
+    """Validationes that all Metric WhereFilters are parseable."""
+
+    @staticmethod
+    @validate_safely(
+        whats_being_done="running model validation ensuring derived metrics properties are configured properly"
+    )
+    def _validate_metric(metric: Metric) -> Sequence[ValidationIssue]:  # noqa: D
+        issues: List[ValidationIssue] = []
+        context = MetricContext(
+            file_context=FileContext.from_metadata(metadata=metric.metadata),
+            metric=MetricModelReference(metric_name=metric.name),
+        )
+
+        if metric.filter is not None:
+            try:
+                metric.filter.call_parameter_sets
+            except Exception as e:
+                issues.append(
+                    generate_exception_issue(
+                        what_was_being_done=f"trying to parse filter of metric `{metric.name}`",
+                        e=e,
+                        context=context,
+                        extras={
+                            "traceback": "".join(traceback.format_tb(e.__traceback__)),
+                            "filter": metric.filter.where_sql_template,
+                        },
+                    )
+                )
+
+        if metric.type_params:
+            measure = metric.type_params.measure
+            if measure is not None and measure.filter is not None:
+                try:
+                    measure.filter.call_parameter_sets
+                except Exception as e:
+                    issues.append(
+                        generate_exception_issue(
+                            what_was_being_done=f"trying to parse the measure filter on metric `{metric.name}`",
```

Comment:
Shall we include the measure name here?
```suggestion
                            what_was_being_done=f"trying to parse the filter for measure `{measure.name}` on metric `{metric.name}`",
```

Reply:
I don't actually think thats necessary. `measure` is directly a `type_params` attribute which is where the filter lives. I worry adding the measure name would actually cause confusion wherein people would look at a measure on a semantic model instead of addressing the measure of the metrics `type_params`

Reply:
Never mind, you're right, I agree 😂